packages/runtime-common/realm-permission-checker.ts

```diff
@@ -1,20 +1,31 @@
+import { MatrixClient } from 'matrix-client';
 import { RealmPermissions } from './realm';
 
 export default class RealmPermissionChecker {
   private realmPermissions: RealmPermissions = {};
+  private matrixClient: MatrixClient;
 
-  constructor(realmPermissions: RealmPermissions) {
+  constructor(realmPermissions: RealmPermissions, matrixClient: MatrixClient) {
     this.realmPermissions = realmPermissions;
+    this.matrixClient = matrixClient;
   }
 
-  can(username: string, action: 'read' | 'write') {
-    return this.getUserPermissions(username).includes(action);
+  async for(username: string) {
+    let isMatrixUserProfileExist = false;
```

Comment:
the english grammar is incorrect here. this should be `doesMatrixUserProfileExist`